lib/src/plugins/html/encoder/delta_html_encoder.dart

```diff
@@ -0,0 +1,119 @@
+import 'dart:convert';
+
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:flutter/material.dart';
+import 'package:html/dom.dart' as dom;
+
+/// A [Delta] encoder that encodes a [Delta] to html.
+///
+/// supported nested styles.
+class DeltaHtmlEncoder extends Converter<Delta, List<dom.Node>> {
+  @override
+  List<dom.Node> convert(Delta input) {
+    final childNodes = <dom.Node>[];
+
+    for (final op in input) {
+      if (op is TextInsert) {
+        final attributes = op.attributes;
+
+        if (attributes != null) {
+          if (attributes.length == 1) {
+            final element = _applyAttributes(attributes, text: op.text);
+            childNodes.add(element);
+          } else {
+            final span = dom.Element.tag(HTMLTags.span);
+            final cssString = _attributesToCssStyle(attributes);
+            if (cssString.isNotEmpty) {
+              span.attributes['style'] = cssString;
+            }
+            span.append(dom.Text(op.text));
+            childNodes.add(span);
+          }
+        } else {
+          childNodes.add(dom.Text(op.text));
+        }
+      }
+    }
+    return childNodes;
+  }
+
+  dom.Element _applyAttributes(Attributes attributes, {required String text}) {
+    if (attributes[AppFlowyRichTextKeys.bold] == true) {
+      final strong = dom.Element.tag(HTMLTags.strong);
+      strong.append(dom.Text(text));
+      return strong;
+    } else if (attributes[AppFlowyRichTextKeys.underline] == true) {
+      final underline = dom.Element.tag(HTMLTags.underline);
+      underline.append(dom.Text(text));
+      return underline;
+    } else if (attributes[AppFlowyRichTextKeys.italic] == true) {
+      final italic = dom.Element.tag(HTMLTags.italic);
+      italic.append(dom.Text(text));
+      return italic;
+    } else if (attributes[AppFlowyRichTextKeys.strikethrough] == true) {
+      final del = dom.Element.tag(HTMLTags.del);
+      del.append(dom.Text(text));
+      return del;
+    } else if (attributes[AppFlowyRichTextKeys.code] == true) {
+      final code = dom.Element.tag(HTMLTags.code);
+      code.append(dom.Text(text));
+      return code;
+    } else if (attributes[AppFlowyRichTextKeys.href] != null) {
+      final anchor = dom.Element.tag(HTMLTags.anchor);
+      anchor.attributes['href'] = attributes[AppFlowyRichTextKeys.href];
+      anchor.append(dom.Text(text));
+      return anchor;
+    } else {
+      final paragraph = dom.Element.tag(HTMLTags.paragraph);
+
+      paragraph.append(dom.Text(text));
+      return paragraph;
+    }
+  }
```

Comment:
I believe this to be slightly more clean.
```suggestion
  dom.Element _applyAttributes(Attributes attributes, {required String text}) {
    dom.Element element;
    if (attributes[AppFlowyRichTextKeys.bold] == true) {
      element = dom.Element.tag(HTMLTags.strong);
    } else if (attributes[AppFlowyRichTextKeys.underline] == true) {
      element = dom.Element.tag(HTMLTags.underline);
    } else if (attributes[AppFlowyRichTextKeys.italic] == true) {
      element = dom.Element.tag(HTMLTags.italic);
    } else if (attributes[AppFlowyRichTextKeys.strikethrough] == true) {
      element = dom.Element.tag(HTMLTags.del);
    } else if (attributes[AppFlowyRichTextKeys.code] == true) {
      element = dom.Element.tag(HTMLTags.code);
    } else if (attributes[AppFlowyRichTextKeys.href] != null) {
      element = dom.Element.tag(HTMLTags.anchor);
      element.attributes['href'] = attributes[AppFlowyRichTextKeys.href];
    } else {
      element = dom.Element.tag(HTMLTags.paragraph);
    }

    element.append(dom.Text(Text));
    return element;
  }
```

Reply:
I replaced it with this function.

```dart
```